### ISSUE TITLE
ACL file is not working on Debian

### ISSUE THREAD

**Issue by IsabelManiega** [eclipse/mosquitto#1399]
Hi everyone,
I create an ACL file on Windows 10, and on Debian as well.
For the case of Debian, the same file is not working corretly. Can I know the problem?
My aclfile:
#users

user John
topic write home/#
topic write subscribe

user Peter
topic write home/#
topic read subscribe

user Mary
topic read home/#

#All users
pattern read home/#

For a user Who is Not John, Peter or Mary (for instance), I get "Connection refused".

When I use as user the added users, It is working (John/ Peter). For Mary does not work correctly.

Thank you in advance.

**Comment by ralight** [comment]
Are you able to share some more of your configuration details and commands you are using here? An ACL should never cause Connection refused, and neither should any of the authentication options actually. When you say that Mary does not work correctly, what do you mean?

What version of Mosquitto are you using?

**Comment by IsabelManiega** [comment]
Thanks you,

I responds your questions:

**Command:** mosquitto_sub -h localhost -p 1883 -t 'test' -u Smith -P hola
Error  Connection refused.

**mosquitto.conf:**
# Place your local configuration in /etc/mosquitto/conf.d/
#
# A full description of the configuration file is at
# /usr/share/doc/mosquitto/examples/mosquitto.conf.example

pid_file /var/run/mosquitto.pid

persistence true
persistence_location /var/lib/mosquitto/

log_dest file /var/log/mosquitto/mosquitto.log

include_dir /etc/mosquitto/conf.d

allow_anonymous false
password_file /etc/mosquitto/passwordfile.txt

**passwordfile.txt:**

Peter:$6$RNT4FJ5pJV3Di02W$u6BKXQzfs4jgjEyeCv/dRguoPl+CBxQ3JXsoz4Ru8nUpof5ZZ89l1fmVkTb1q4c6rOTbd+6ZsiE/FPNaGdAHeA==
John:$6$UUaWitKY2z3HoJTr$SnnI+bpkAXchJObonxbWsIs3W6lAc3HDLlmXXBUOQ3sGKsiAqPZwIJqKLYbREjZO/S8eSXYCnqyzvlawn7bw7w==
Mary:$6$w5skuhElMkpO84qo$Tl8Oo5BJ74lHZgxqEolCGhU2w57ZPAC70uY1rO+ZMlSvKGO9K26pgWXehkZSEEYlxK+XfL548YvNaBsIjiQIzg==
Smith:$6$krCYk4sn+xUBtnq6$Bec4bwbe/ARDORKqibg91gppGat2SV2O54whx2AhPk6K4fT4abqBrarodjeCDR1ypWrZ9SqeMYeED75KPfV/ww==

Mary subscribes and publishes as well.

**Comment by ralight** [comment]
Right, I think I understand what is happening here, and I apologise that I misunderstood at the beginning.

I believe you simply have an authentication problem, it's nothing to do with the ACL. Using the password file you gave, I also get `Connection Refused`. If I use a more recent version of mosquitto_sub, I get the slightly more useful message `Connection Refused: not authorised.`

I then tried updating the password file:

`mosquitto_passwd passwordfile.txt Smith`

and used `hola` as the password. Repeating the mosquitto_sub command you gave above then worked.

**Comment by IsabelManiega** [comment]
Firstly, thank you so much for your answer,
It is possible that this is the problem, I Will work on It, but i have another problem.
When I add a new user in the password file, I need to restart the mosquitto:
sudo systemctl restart mosquitto
I have tried to restore the configuration:
mosquitto -c /etc/mosquitto/mosquitto.conf
(Without restarting Mosquitto), 
but It is not working.
I do not know if there is another method to add users...

**Comment by squawk979** [comment]
Had the exact same aggravating problem.  
Eventually found upgrading from 1.4.8 -> 1.6.9 resolved the issue for me.
